src/main/Main.hs

```diff
@@ -779,6 +788,68 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
           pkgopts <- getPkgOpts menv wc pkgs
           return (prefix ++ compilerExeName wc, pkgopts ++ args)
 
+scriptOptsParser :: Parser ([String], [String])
+scriptOptsParser = (,)
+    <$> many (strOption (long "package" <> help "Additional packages that must be installed"))
+    <*> many (strArgument (metavar "-- ARGS (e.g. stack ghc -- X.hs -o x)"))
+
+-- | Run a Stack Script
+scriptCmd :: ([String], [String]) -> GlobalOpts -> IO ()
+scriptCmd (packages', args') go' = do
+    let go = go'
+            { globalConfigMonoid = (globalConfigMonoid go')
+                { configMonoidInstallGHC = First $ Just True
+                }
+            }
+    withBuildConfigNoFileAndLock go $ \lk -> do
+        config <- asks getConfig
+        menv <- liftIO $ configEnvOverride config defaultEnvSettings
+        wc <- getWhichCompiler
+
+        let targets = concatMap wordsComma packages'
+            targetsSet = Set.fromList targets
+
+        -- Ensure only package names are provided. We do not allow
+        -- overriding packages in a snapshot.
+        mapM_ parsePackageNameFromString targets
+
+        unless (null targets) $ do
+            -- Optimization: use the relatively cheap ghc-pkg list
+            -- --simple-output to check which packages are installed
+            -- already. If all needed packages are available, we can
+            -- skip the (rather expensive) build call below.
+            bss <- sinkProcessStdout
+                Nothing menv (ghcPkgExeName wc)
+                ["list", "--simple-output"] CL.consume -- FIXME use the package info from envConfigPackages, or is that crazy?
```

Comment:
RE the FIXME, I don't think that's necessary - envConfigPackages is for local packages.  If there is no settings file, then there can't be local packages.